### PR TITLE
[Core][Geometry] Enable Creation of IntegrationPoints and QuadraturePointGeometries from Geometry

### PR DESCRIPTION
**Pull request by tteschemacher**
This PR allows to create integration points and quadrature point geometries from coupling geometries. This allows to couple generically geometries of different types. This is not only for nurbs geometries.

Furthermore, it updates the possibilities to interact with the CreateIntegrationPoints with the IntegrationInfo(). This helps to generalize it for many applications.

The corresponding function to IsInsideLocalSpace, the SetInsideLocalSpace is added to the geometry, accordingly.

### CODE REVIEW

File: kratos/utilities/tessellation_utilities/tessellation.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    ///@}
+    ///@name Constructors
+    ///@{
+
+    virtual void Tessellate(
+        const GeometryType& rGeometry,
+        const double Tolerance,
+        const int NumberOfGuessesPerInterval = 1)
+    {
+        KRATOS_ERROR << "Calling Tessellate from base Tessellation class."
```
Comment (loumalouomega) [inline kratos/utilities/tessellation_utilities/tessellation.h:51]: Are you going to consider the Delaunator utilities  (https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/delaunator_utilities.cpp) in a derived class?

**Review by loumalouomega — COMMENTED** [review]
Alright, but I am not part of the @KratosMultiphysics/technical-committee , so I will not approve

**Comment by loumalouomega** [comment]
CI is failing

**Comment by loumalouomega** [comment]
> CI is failing

@tteschemacher Basically the member variables must be constructed in the proper order. This is usuallya  warning, now transformed into error

**Comment by loumalouomega** [comment]
> > CI is failing
> 
> @tteschemacher Basically the member variables must be constructed in the proper order. This is usuallya warning, now transformed into error

~~~
 /__w/Kratos/Kratos/kratos/integration/integration_info.h:262:14: error: 'Kratos::IntegrationInfo::mNumberOfIntegrationPointsPerSpan' will be initialized after [-Werror=reorder]
  262 |     SizeType mNumberOfIntegrationPointsPerSpan;
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/Kratos/Kratos/kratos/integration/integration_info.h:260:22: error:   'Kratos::IntegrationInfo::QuadratureMethod Kratos::IntegrationInfo::mQuadratureMethod' [-Werror=reorder]
  260 |     QuadratureMethod mQuadratureMethod;
      |                      ^~~~~~~~~~~~~~~~~
/__w/Kratos/Kratos/kratos/integration/integration_info.h:76:5: error:   when initialized here [-Werror=reorder]
   76 |     IntegrationInfo(SizeType NumberOfIntegrationPointsPerSpan,
~~~

**Comment by loumalouomega** [comment]
@tteschemacher I fixed it

File: kratos/utilities/tessellation_utilities/tessellation.h (hunk 1 above)
Comment (tteschemacher) [reply kratos/utilities/tessellation_utilities/tessellation.h:51]: Yes, I actually want to take this into account for 2D.

**Comment by tteschemacher** [comment]
> 
> 
> @tteschemacher I fixed it

Thanks

File: kratos/utilities/tessellation_utilities/tessellation.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/utilities/tessellation_utilities/tessellation.h:51]: :+1:

**Comment by loumalouomega** [comment]
@tteschemacher still fails, now it is a different issue

**Comment by loumalouomega** [comment]
> @tteschemacher still fails, now it is a different issue

Passing const values as non const (or viceversa)

**Comment by tteschemacher** [comment]
@loumalouomega do you think we could create an interface to use your developments with this interface. Theoretically, all information for coupling/ contact is provided here.

**Comment by tteschemacher** [comment]
This PR is not needed anymore as all changes are now within #8596 .

**Pull request closed without merging**